### PR TITLE
Allow depending to a @task_group as a whole

### LINKED ISSUE #19903: Error setting dependencies on task_group defined using the decorator
### Apache Airflow version

2.2.2 (latest released)

### Operating System

MacOS 11.6.1

### Versions of Apache Airflow Providers

$ pip freeze | grep airflow
apache-airflow==2.2.2
apache-airflow-providers-celery==2.1.0
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-http==2.0.1
apache-airflow-providers-imap==2.0.1
apache-airflow-providers-sqlite==2.0.1

### Deployment

Other

### Deployment details

`airflow standalone`

### What happened

```
AttributeError: 'NoneType' object has no attribute 'update_relative'
```

### What you expected to happen

Task group should be set as downstream of `start` task, and upstream of `end` task

### How to reproduce

* Add the following code to dags folder

```python
from datetime import datetime

from airflow.decorators import dag, task, task_group


@dag(start_date=datetime(2023, 1, 1), schedule_interval="@once")
def test_dag_1():
    @task
    def start():
        pass

    @task
    def do_thing(x):
        print(x)

    @task_group
    def do_all_things():
        do_thing(1)
        do_thing(2)

    @task
    def end():
        pass

    start() >> do_all_things() >> end()


test_dag_1()
```

* Run `airflow standalone`

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by uranusjr**
Fix #19903.

Instead of always using the task returned by the task group function as the dependency representation, also allow a task group function to *not* return any task specifically.

Setting dependency to such a group would depend on the group's contained tasks as a whole, and not any specific task in it.

This renders this DAG definition taken from https://github.com/apache/airflow/issues/19903#issuecomment-1004105943 (which would fail to parse currently)

```python
@dag(start_date=datetime(2023, 1, 1), schedule_interval="@once")
def test_dag_1():
    @task
    def start():
        pass

    @task
    def do_thing(x):
        print(x)

    @task_group
    def do_all_things():
        do_thing(1)
        do_thing(2)

    @task
    def end():
        pass

    start() >> do_all_things() >> end()

my_dag = test_dag_1()
```

like this:

image

~~Tests to come later.~~ Done

### CODE REVIEW

File: airflow/decorators/task_group.py [hunk 1]
```diff
@@ -92,9 +92,23 @@ def factory(*args, **kwargs):
                 *task_group_bound_args.args,
                 add_suffix_on_collision=True,
                 **task_group_bound_args.kwargs,
-            ):
+            ) as task_group:
                 # Invoke function to run Tasks inside the TaskGroup
-                return f(*args, **kwargs)
+                f_retval = f(*args, **kwargs)
+
+            # If 'f' returns a task, bind dependencies to it. Equilvanet to
```
Comment (josh-fell) [inline airflow/decorators/task_group.py]:
```suggestion
            # If 'f' returns a task, bind dependencies to it. Equivalent to
```

**Review by josh-fell — COMMENTED** [review]
Love that this gets to parity with the context manager version of TaskGroups!

**Review by eladkal — APPROVED** [review]
Nice!

**Comment by josh-fell** [comment]
Doesn't need to be in this PR but it might be nice to have some documentation on these two behaviors with the `@task_group` decorator. 

@uranusjr I can log a request for this if you don't want to take this up. Let me know.

**Comment by uranusjr** [comment]
I created #20762 to track documentation changes.

**Pull request merged**